File: backend/src/main/java/com/apocalypse/thefall/dto/item/ArmorDamageDto.java
package com.apocalypse.thefall.dto.item;

import com.apocalypse.thefall.dto.character.stats.DataItemDto;
import lombok.Builder;

@Builder
public record ArmorDamageDto(
        Long id,
        DataItemDto damageType,
        Integer threshold,
        Integer resistance
) {
}

File: backend/src/main/java/com/apocalypse/thefall/dto/item/ArmorDamageDto.kt
package com.apocalypse.thefall.dto.item

import com.apocalypse.thefall.dto.character.stats.DataItemDto

@JvmRecord
data class ArmorDamageDto(
    val id: Long?,
    val damageType: DataItemDto?,
    val threshold: Int?,
    val resistance: Int?
) 